templates/common/on-prem/files/NetworkManager-resolv-prepender.yaml

```diff
@@ -23,12 +23,12 @@ contents:
 
     export DHCP6_FQDN_FQDN IP4_DOMAINS IP6_DOMAINS
     export -f resolv_prepender
-    # Given an overall Network Manager dispatcher timeout of 90 seconds, and 3 events which may occur within
-    # this time period (up, dhcp4, dhcp6), we must enforce a time limit for each event.
+    # Given an overall Network Manager dispatcher timeout of 90 seconds, and multiple events which
+    # may occur within this time period, we must enforce a time limit for each event.
     case "$STATUS" in
-      up|dhcp4-change|dhcp6-change)
+      up|dhcp4-change|dhcp6-change|reapply)
         >&2 echo "NM resolv-prepender triggered by ${IFACE} ${STATUS}."
-        if ! timeout 30s bash -c resolv_prepender; then
+        if ! timeout 20s bash -c resolv_prepender; then
```

Comment:
I wonder if we need to do this. Can you get both an up and reapply status in the same activation? It seems like those should be mutually exclusive.